README.md
# Flutter Numeral System Both in International and Indian/ Hindu-Arabic System

This plugin helps you to display number as per International or Indian Numeral System. Internation Numeral System means the system that counts number in the form of ONES, TENS, HUNDREDS, THOUSANDS, TEN THOUSANDS, HUNDRED THOUSANDS, MILLION, etc. Indian or Hindu-Arabic System means the system that counts number in the form of ONES, TENS, HUNDREDS, THOUSANDS, TEN THOUSANDS, LAKH , TEN LAKH, CRORE, etc.

## Table of contents

- [Features](#features)
- [Installation](#installation)
- [Support and feedback](#support-and-feedback)


## Features

* You can use any of the numeral system.
* You can add textstyle.
* You can choose the numbers of digit to be displayed after decimal.


## Installation

Add package to the the pubspec.yaml.

```dart
numeral_system:
```

Import the package.

```dart
import 'package:numeral_system/numeral_system.dart';
```

Add NumeralSystem to the widget tree

```dart
Scaffold(
    body: Center(
        child:
            //By Default i.e INTERNATIONAL and two digits after decimal
            NumeralSystem(digit: 934023),
        )
    )
```

For Indian System

```dart
NumeralSystem(
    numberSystem: NumberSystem.indian,
    digit: 987387659876,
    digitAfterDecimal: DigitAfterDecimal.one,
    ),
```

For 3 digits after decimal

```dart
NumeralSystem(
    digit: 195659876,
    numberSystem: NumberSystem.indian,
    digitAfterDecimal: DigitAfterDecimal.three,
    ),
```

For adding textstyle

```dart
 NumeralSystem(
    digit: 987654,
    digitAfterDecimal: DigitAfterDecimal.three,
    textStyle: TextStyle(
        fontSize: 20,
        fontWeight: FontWeight.bold,
        color: Colors.red)),
    ),
```


## Support and feedback

If I had missed anything to add, feel free to open a ticket or contribute!
You can reach me here:
[LinkedIn](https://www.linkedin.com/in/lakshydeep-14/) |
[GitHub](https://github.com/lakshydeep-14) |
[Medium](https://lakshydeep-14.medium.com/) |
[Google DevLibrary](https://devlibrary.withgoogle.com/authors/lakshydeep-14)


CHANGELOG.md
## 2.0.0+1

* Docs updated

## 2.0.0

* Max 3 digits after decimal were added in a feasible way.

## 0.0.1

* TODO: Describe initial release.


lib/numeral_system.dart
library numeral_system;

import 'package:flutter/material.dart';
// Numeral System type
// International means ONES, TENS, HUNDREDS, THOUSANDS, TEN THOUSAND, HUNDRED THOUSAND, MILLIONS, TEN MILLION, and so on.
// Indian means ONES, TENS, HUNDRED, THOUSAND, TEN THOUSAND, LAKH, TEN LAKH, CRORE, TEN CRORE, and so on.

enum NumberSystem { indian, international }

enum DigitAfterDecimal {
  one,
  two,
  three,
}

// class for numneral system
class NumeralSystem extends StatefulWidget {
  // number system type, either INTERNATIONAL or INDIAN
  // By default, it's INTERNATIONAL.
  final NumberSystem numberSystem;

  // the number which you want to give
  final num digit;

  // textstyle
  // By default it's color is black
  final TextStyle textStyle;

  //int value to show number of digits after decimal
  //By default it's 3 which mean it will show 2 digits after decimal.
  final DigitAfterDecimal digitAfterDecimal;

  const NumeralSystem(
      {super.key,
      this.numberSystem = NumberSystem.international,
      required this.digit,
      this.digitAfterDecimal = DigitAfterDecimal.two,
      this.textStyle = const TextStyle(color: Colors.black)});

  @override
  State<NumeralSystem> createState() => _NumeralSystemState();
}

class _NumeralSystemState extends State<NumeralSystem> {
  @override
  Widget build(BuildContext context) {
    return Text(
      widget.numberSystem == NumberSystem.indian
          ? _indian(widget.digit)
          : _international(widget.digit),
      style: widget.textStyle,
    );
  }

  String getNumber(double input) =>
      "${input.toString().split(".").first}.${input.toStringAsFixed(4).toString().split(".").last.substring(0, widget.digitAfterDecimal == DigitAfterDecimal.one ? 1 : widget.digitAfterDecimal == DigitAfterDecimal.two ? 2 : 3)}";
  _indian(num number) {
    if (number == 0) {
      return '0';
    } else if (number <= 999) {
      return number.toString();
    } else if (number > 999 && number <= 99999) {
      return "${getNumber((number / 1000))} K";
    } else if (number > 99999 && number <= 9999999) {
      return "${getNumber((number / 100000))} L";
    } else if (number > 9999999 && number <= 999999999) {
      return "${getNumber((number / 10000000))} Cr";
    } else {
      return "${getNumber((number / 1000000000))} Ar";
    }
  }

  _international(num number) {
    if (number == 0) {
      return '0';
    } else if (number <= 999) {
      return number.toString();
    } else if (number > 999 && number <= 999999) {
      return "${getNumber((number / 1000))} K";
    } else if (number > 999999 && number <= 999999999) {
      return "${getNumber((number / 1000000))} M";
    } else if (number > 999999999 && number <= 999999999999) {
      return "${getNumber((number / 1000000000))} B";
    } else {
      return "${getNumber((number / 1000000000000))} T";
    }
  }
}


example/lib/main.dart
// ignore_for_file: prefer_const_constructors, prefer_const_literals_to_create_immutables

import 'package:flutter/material.dart';
import 'package:numeral_system/numeral_system.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(home: Demo());
  }
}

class Demo extends StatefulWidget {
  const Demo({super.key});

  @override
  State<Demo> createState() => _DemoState();
}

class _DemoState extends State<Demo> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Numeral System"),
        centerTitle: true,
      ),
      body: Center(
        child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              //By Default i.e INTERNATIONAL
              NumeralSystem(digit: 12345000),

              //ADDING TEXTSTYLE
              NumeralSystem(
                  digit: 987654,
                  digitAfterDecimal: DigitAfterDecimal.three,
                  textStyle: TextStyle(
                      fontSize: 20,
                      fontWeight: FontWeight.bold,
                      color: Colors.red)),

              // After decimal 2 digits
              NumeralSystem(
                numberSystem: NumberSystem.indian,
                digit: 987387659876,
                digitAfterDecimal: DigitAfterDecimal.two,
              ),

              //3 digits
              NumeralSystem(
                digit: 195659876,
                numberSystem: NumberSystem.indian,
                digitAfterDecimal: DigitAfterDecimal.three,
              ),
            ]),
      ),
    );
  }
}
